src/modifier/default.rs

```diff
@@ -313,13 +323,358 @@ fn div_numeric(map: &LinearMap<Atom, Expression, 8>) -> (Expression, bool) {
+            ),
+            Expression::Power(e1, e2) => Expression::Power(
+                Box::new(value_replace(e1, var, val)),
+                Box::new(value_replace(e2, var, val)),
+            ),
+            Expression::Modulus(e1, e2) => Expression::Modulus(
+                Box::new(value_replace(e1, var, val)),
+                Box::new(value_replace(e2, var, val)),
+            ),
+
+            Expression::Function { name, args } => Expression::Function {
+                name: name.clone(),
+                args: args
+                    .iter()
+                    .map(|e| Box::new(value_replace(e, var, val)))
+                    .collect(),
+            },
+            Expression::Vector { backing, size } => Expression::Vector {
+                backing: backing
+                    .iter()
+                    .map(|e| Box::new(value_replace(e, var, val)))
+                    .collect(),
+                size: *size,
+            },
+            Expression::Matrix { backing, shape } => Expression::Matrix {
+                backing: backing
+                    .iter()
+                    .map(|e| Box::new(value_replace(e, var, val)))
+                    .collect(),
+                shape: *shape,
+            },
+        }
+    }
+}
+
+//approximate the derivative of a function
+fn diff_approx(map: &LinearMap<Atom, Expression, 8>) -> (Expression, bool) {
+    let expr = map.get(&Atom::Escape('*', 1)).unwrap();
+    let var = map.get(&Atom::Escape('*', 2)).unwrap();
+    let val = map.get(&Atom::Escape('A', 1)).unwrap();
+
+    let h = 0.0001;
+
+    match val {
+        Expression::Atom(Atom::Numeric(n)) => (
+            //symmetric difference quotient at h = 0.0001
+            Expression::Divide(
+                Box::new(Expression::Subtract(
+                    Box::new(value_replace(
+                        expr,
+                        var,
+                        &Expression::Atom(Atom::Numeric(*n + Numeric::Decimal(h))),
+                    )),
+                    Box::new(value_replace(
+                        expr,
+                        var,
+                        &Expression::Atom(Atom::Numeric(*n - Numeric::Decimal(h))),
+                    )),
+                )),
+                Box::new(Expression::Atom(Atom::Numeric(Numeric::Decimal(2.0 * h)))),
+            ),
+            true,
+        ),
+        _ => (
+            Expression::Atom(Atom::Error(crate::Error::InvalidSyntax)),
+            false,
+        ),
+    }
+}
+
+//approximate the integral of a function
+fn int_approx(map: &LinearMap<Atom, Expression, 8>) -> (Expression, bool) {
+    let expr = map.get(&Atom::Escape('*', 1)).unwrap();
+    let var = map.get(&Atom::Escape('*', 2)).unwrap();
+    let val1 = map.get(&Atom::Escape('A', 1)).unwrap();
+    let val2 = map.get(&Atom::Escape('A', 2)).unwrap();
+
+    //TODO: use a more accurate approximation
+    //trapeziodal rule: (b-a)*((f(a)+f(b))/2)
```

Comment:
Wouldn't it be better to leave the "approximation" of a numerical integral to the evaluation portion?